lambdas/process-journey-event/src/main/java/uk/gov/di/ipv/core/processjourneyevent/ProcessJourneyEventHandler.java

```diff
@@ -245,4 +265,46 @@ private Map<IpvJourneyTypes, StateMachine> loadStateMachines(
         }
         return stateMachinesMap;
     }
+
+    @Tracing
+    private void checkIfStartingMitigationJourney(
+            Map<String, String> input,
+            StepResponse stepResponse,
+            ClientOAuthSessionItem clientOAuthSessionItem)
+            throws HttpResponseExceptionWithErrorBody, SqsException {
+
+        String ipvSessionId = StepFunctionHelpers.getIpvSessionId(input);
+        String ipAddress = StepFunctionHelpers.getIpAddress(input);
+
+        Boolean mitigationStart = getMitigationStart(stepResponse);
+        if (Objects.nonNull(mitigationStart) && mitigationStart) {
+            sendAuditEvent(ipvSessionId, ipAddress, clientOAuthSessionItem);
+        }
+    }
+
+    private Boolean getMitigationStart(StepResponse stepResponse) {
+        if (stepResponse instanceof PageStepResponse) {
+            return ((PageStepResponse) stepResponse).getMitigationStart();
+        } else if (stepResponse instanceof ProcessStepResponse) {
+            return ((ProcessStepResponse) stepResponse).getMitigationStart();
+        }
+        return false;
+    }
```

Comment:
You can cast and assign in a single expression to simplify this. You might also be able to use one of a few tricks to simplify the boolean check (here are two options, you can decide which you think is cleaner...)
```
if (stepResponse instanceof PageStepResponse pageStepResponse) {
    return Objects.requireNonNullElse(pageStepResponse.getMitigationStart(), false);
} else if (stepResponse instanceof ProcessStepResponse processStepResponse) {
    return Boolean.TRUE.equals(processStepResponse.getMitigationStart());
}

return false;
```

Incidentally, if we were using Java 21, this could be a neat use for a pattern matching expression. Something like:
```
return Boolean.TRUE.equals(switch (stepResponse) {
    case PageStepResponse page -> page.getMitigationStart();
    case ProcessStepResponse process -> process.getMitigationStart();
    default -> null;
});
```

Reply:
You could also consider whether you want to limit `mitigationStart` to just `page` and `process` states - if it applies to everything then you can add it to the interface and no casting is needed :)